cpmpy/expressions/globalconstraints.py

```diff
@@ -291,6 +299,52 @@ def value(self):
         rev = [argval(a) for a in self.args[1]]
         return all(rev[x] == i for i, x in enumerate(fwd))
 
+class InverseOne(GlobalConstraint):
+    """
+       Inverse (aka channeling / assignment) constraint but with only one array.
```

Comment:
Which constraint from the global constraint catalog does this correspond to? Why does it need a separate class?

Reply:
the Channel(X) from the xcsp3 competition where X[i] =j <-> X[j]=i
New global because I ask (CFR slack conversation) and wout told me to do an other global instead of just extending Inverse(X,X) as not all the solvers accepts twice the same var as args